controllers/nginx/pkg/cmd/controller/nginx.go

```diff
@@ -92,15 +93,48 @@ type NGINXController struct {
 // Start start a new NGINX master process running in foreground.
 func (n NGINXController) Start() {
 	glog.Info("starting NGINX process...")
+
+	done := make(chan error, 1)
 	cmd := exec.Command(n.binary, "-c", cfgPath)
+	n.start(cmd, done)
+	select {
+	case err := <-done:
+		if exitError, ok := err.(*exec.ExitError); ok {
```

Comment:
please explain why this is needed (i.e why is it important to wait for master process in this way vs just running start and assuming the liveness check will fail if the master never comes up?)